File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>m;
        int sum=0;
        int res=0;
        for(int i=0;i<nums.size();i++)
        {
            sum+=nums[i];
            if(sum==k)
                res++;
            if(m.find(sum-k)!=m.end())
                res+=m[sum-k];
            m[sum]++;
        }
        return res;
    }
};

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans=0;
        prefSum=0;
        d={0:1};
        for num in nums:
            prefSum+=num;
            if (prefSum-k) in d:
                ans+=d[prefSum-k];
                
            if (prefSum) not in d:
                d[prefSum]=1;
            else:
                d[prefSum]+=1;
        return ans;
        